src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserHelperClasses.cs

```diff
@@ -240,13 +242,176 @@ sealed internal class SqlEnvChange
         internal int length;
         internal string newValue;
         internal string oldValue;
+        /// <summary>
+        /// contains binary data, before using this field check newBinRented to see if you can take the field array or whether you should allocate and copy
+        /// </summary>
         internal byte[] newBinValue;
+        /// <summary>
+        /// contains binary data, before using this field check newBinRented to see if you can take the field array or whether you should allocate and copy
+        /// </summary>
         internal byte[] oldBinValue;
         internal long newLongValue;
         internal long oldLongValue;
         internal SqlCollation newCollation;
         internal SqlCollation oldCollation;
         internal RoutingInfo newRoutingInfo;
+        internal bool newBinRented;
+        internal bool oldBinRented;
+
+        public SqlEnvChange Next;
```

Comment:
Let's make all these public APIs internal.

Reply:
Done, and I applied the class modifier guidelines to the file as well.